sphinx/ext/rose_domain.py

```diff
@@ -0,0 +1,741 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2012-7 Met Office.
+#
+# This file is part of Rose, a framework for meteorological suites.
+#
+# Rose is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Rose is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Rose. If not, see <http://www.gnu.org/licenses/>.
+# -----------------------------------------------------------------------------
+"""The rose domain is for documenting rose configurations and built-in
+applications.
+
+Rose Objects:
+    The rose domain supports the following object types:
+
+    * Applications using the prefix ``app`` (i.e. ``rose:app``).
+    * Configuration files using the prefix ``file`` (i.e. ``rose:file``).
+    * Configuration sections using the prefix ``conf`` (i.e. ``rose:conf``).
+    * Configuration settings using the prefix ``conf`` (i.e. ``rose:conf``).
+
+    See the corresponding directives for more information on each object type.
+
+Reference Syntax:
+    Once documented objects can be referenced using the following syntax:
+
+    ``:rose:CONFIG-FILE.[parent-section]child-config``
+
+    Where ``CONFIG-FILE`` is:
+
+    * ``APP-NAME`` for applications (e.g. ``fcm_make``).
+    * ``FILE-NAME`` for configuration files (e.g. ``rose.conf``).
+
+Referencing From RST Files:
+    To reference a rose object add the object ``TYPE`` into the domain
+    (e.g. ``conf`` or ``app``).
+
+    .. code-block:: rst
+
+       :rose:TYPE:`CONFIG-FILE.[parent-section]child-config`
+
+    e.g:
+
+    .. code-block:: rst
+
+       * :rose:app:`fcm_make`
+       * :rose:conf:`fcm_make.args`
+
+Example:
+    .. code-block:: rst
+
+       .. rose:file:: rose-suite.conf
+
+          The config file used for configuring suite level settings.
+
+          .. rose:conf:: jinja2:suite.rc
+
+             A section for specifying Jinja2 settings for use in the ``suite.rc``
+             file.
+
+             Note that one ommits the square brackets for config sections, if
+             :rose:conf: contains other :rose:conf:`s then it is implicitly a
```

Comment:
The pygments RST lexer cannot handle domains, note this documentation is for developers only and is hidden on the main contents page.

Reply:
I know, but given it you spent all that time writing it I thought it might as well be worth a polish. But I can leave it if you really think it is not worth bothering with.

Reply:
If you think it isn't worth it, fair enough. But you must have spent a bit of time writing it so changing a few little things wouldn't seem unreasonable to me...